### PR TITLE
`kedro-datasets` CI setup

### PR DESCRIPTION
**Pull request by noklam**
Signed-off-by: Nok Chan <nok.lam.chan@quantumblack.com>

## Development notes

- [ ] ~~Enable auto-release process~~  Scope of another PR
- [x] Add CI/CD setup for `kedro-datasets` and refactor it since `kedro-datasets` carry more dependencies that will slow down other plugins significantly.
- [ ] Bonus: Optimize the CI  a little bit since other plugins need much fewer dependencies, it's not too important since most plugins aren't updated that frequently.

`circleci config validate continue_config.yml` did save me some time from not pushing invalid config and realise it later.

## Testing
- [x] Manually Test changes will trigger the CI, tested at [a55ff3d](https://github.com/kedro-org/kedro-plugins/pull/40/commits/a55ff3df03f1e09317b3c790b4858e4de8de5aae)

## Checklist
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: .circleci/continue_config.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: 2.1
 
 orbs:
-  win: circleci/windows@2.4.0
+  win: circleci/windows@2.4.1
```
Comment (noklam) [inline .circleci/continue_config.yml:4]: just to align with `kedro` repo

File: .circleci/continue_config.yml [hunk 2]
```diff
@@ -174,19 +237,66 @@ jobs:
       - checkout
       - win_setup_conda:
           python_version: <<parameters.python_version>>
+      - when:
+          python_version: <<parameters.python_version>>
+          condition:
+            equal: [true, <<pipeline.parameters.run-build-kedro-datasets>>]
```
Comment (noklam) [inline .circleci/continue_config.yml:243]: Skipping the slow install for plugins other then `kedro-datasets`

**Comment by noklam** [comment]
The unit test is failing because our tests actually cover 97% of `CSVDataSet`! I run the same tests in `kedro` it's the same result. So there are some unknown tests covering the remaining 3% outside `tests/datasets/*.py`, I haven't found it yet.

**Comment by MerelTheisenQB** [comment]
> The unit test is failing because our tests actually cover 97% of `CSVDataSet`! I run the same tests in `kedro` it's the same result. So there are some unknown tests covering the remaining 3% outside `tests/datasets/*.py`, I haven't found it yet.

Interesting... could it maybe be in the `data_catalog` tests? It looks like we use `CSVDataSet` there quite a lot.

**Review by MerelTheisenQB — APPROVED** [review]
This setup looks good to me 👍 

And TIL `circleci config validate continue_config.yml ` 😄

**Comment by noklam** [comment]
Thanks @MerelTheisenQB 

image


TIL You can use `--cov-context` which tells you which exact test runs which lines of code.
```python
 pytest tests/io/test_incremental_dataset.py  --cov-config pyproject.toml --cov-context=test
coverage html --show-contexts
```

**Pull request merged**